### PR TITLE
needle 0.16.2.2

### PR DESCRIPTION
**Pull request by tianskylan**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by tianskylan** [comment]
CI is complaining about the following:
```
==> FAILED
Error: 1 problem in 1 formula detected
needle:
  * Non-executables were installed to "/usr/local/opt/needle/bin"
    The offending files are:
      /usr/local/opt/needle/bin/lib_InternalSwiftSyntaxParser.dylib
```
We keep that `lib_InternalSwiftSyntaxParser.dylib` file next to our `needle` binary as a supporting library; not intending to execute it directly.
Is there a way to silence this CI error?

**Comment by SMillerDev** [comment]
No, that file should not be in that directory because it'll mean that it's in `PATH` for people.

File: Formula/needle.rb [hunk 1]
```diff
@@ -12,11 +12,14 @@ class Needle < Formula
...
   end
 
-  depends_on xcode: ["11.3", :build]
+  depends_on xcode: ["11.6", :build]
   depends_on xcode: "6.0"
 
   def install
-    system "make", "install", "BINARY_FOLDER_PREFIX=#{prefix}"
+    system "make", "archive_generator"
+    system "install_name_tool", "-change", "@executable_path/lib_InternalSwiftSyntaxParser.dylib", "@executable_path/../libexec/lib_InternalSwiftSyntaxParser.dylib", "./Generator/bin/needle"
```
Comment (SMillerDev) [inline Formula/needle.rb:20]: This really needs to be configurable upstream, not patched in the homebrew formula

Comment (tianskylan) [reply Formula/needle.rb:20]: Thanks for the review. By `upstream` do you mean to handle things in our repo instead (i.e. Needle's Makefile)?

Comment (SMillerDev) [reply Formula/needle.rb:20]: Yes, the install location of the library should be configurable to allow users (like brew) to follow best-practices.

File: Formula/needle.rb [hunk 2]
```diff
@@ -12,11 +12,13 @@ class Needle < Formula
...
   end
 
-  depends_on xcode: ["11.3", :build]
+  depends_on xcode: ["11.6", :build]
   depends_on xcode: "6.0"
 
   def install
     system "make", "install", "BINARY_FOLDER_PREFIX=#{prefix}"
+    bin.install "./Generator/bin/needle"
+    libexec.install "./Generator/bin/lib_InternalSwiftSyntaxParser.dylib"
```
Comment (SMillerDev) [inline Formula/needle.rb:21]: The library doesn't need to be next to the executable to function?

Comment (tianskylan) [reply Formula/needle.rb:21]: Nope, I've configured it in `make install` so that it could find the .dylib in relative path.

**Review by SMillerDev — APPROVED** [review]
Thanks @tianskylan  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**